_posts/2018-06-25-java-releases.md

```diff
@@ -0,0 +1,85 @@
+newest Java version. Even if this period was to short based on some constraints it was possible to extend it 
+by buying commercial support for a Java version.
+
+As already said all this points will change in future based on that the new release train for Java. The 
+following image shows the release train as it was announced by Oracle.
+
+XXXXXX BILD XXXXXXXX
+
+Starting with Java 9 mostly all Java version will only have a lifetime of 6 month. After this periode no more 
+updates will be provided for the version. Even if you have a commercial contract with Oracle you won’t get 
+any additional updates for Java 9 or 10 after this 6 months. Next to this all XX years a LTS ( long term 
+support) version of Java will be released. The first version with LTS support will be Java 11 that is 
+scheduled for this autumn. This LTS version will provide a commercial support with updates for XX years. The 
+most important fact is that updates for LTS version can only be accessed and used by people that buy 
+commercial support at Oracle. Based on my daily work I would say that most companies that create software 
+based on Java do not have commercial support today since the release train of Java was quite „friendly“ and 
+an adoption of the newest version was easily possible.
+
+## Why is Oracle doing this?
+TODO: ASChneller neue Features, Geld verdienen
+
+## What does the new release train mean to my company?
+If we concentrate on the Oracle JDK the answer to that question is quite easy and you can choose between 3 
+options:
+- Update the Java version all 6 month. By doing so you will always depend on the supported Java version and 
+will automatically get all important feature and security updates.
+- Buy commercial support from Oracle and migrate only from one LTS version to the next LTS version. This 
+would mean that you migrate from Java 8 to Java 11 maybe within the next year and than from Java 11 to Java 
+XX in XXXX.
+- Stay on a Java version that is not supported anymore. Once the free support of a Java version ends nobody 
+forbids you to use the unsupported version in future. By doing so you do not need to pay commercial support 
+or get into a maybe stressfull migration all 6 month. While you can now easily decide on your own when you 
+want to migrate to a newer Java version you will miss security updates and bugfixes. While a bugfix is not 
+really critical as long as the bug do not affect your software, open security issues can end in horrible 
+problems.
+While this option provides several strategies and mostly at least one of them can be adopted to the general 
+workflow of a developer team you should have one important point in mind: One of the main reasons why Oracle 
+has chosen this new release train is to offer new languages features faster. When generics, annotations or 
+lambdas were introduced in Java the developer community had quite a lot of time to understand and adopt that 
```

Comment:
When new language features are introduced in Java they take a considerable amount of time until they are adopted and widely used in the daily work. As an example both the generic (introduced with Java 5) as well as lambdas (introduced with Java 8) have only slowly made their progress to the community. Part of this slow adoption speed is due to the countless open source frameworks and libraries which are a core part of the Java ecosystem. A new language feature will not get adopted until it can inter operate with the dependencies of an application.

Because of this it was a common pattern to stay one or two versions behind the newest release to be in the sweet spot of having all your dependencies support your Java version as well as still getting security updates from Oracle. Also the libraries which are mostly developed in the spear time and with limited resources did not feel too much pressure to keep up with the newest Java version as they could relay on the long support by Oracle. So most of them invested the resources in new features instead of migrations.

Reply:
@sclassen I do not think that the 2nd paragraph is right. Based on that you would prefer to use Java 7 today...

Reply:
I think right now Java 8 is the sweet spot. I haven't seen any project on Java 9...
But I do not want to write your article. It is all just a suggestion.